# Generated by Django 5.2 on 2025-05-03 09:36

import django.core.validators
from django.db import migrations, models


class Migration(migrations.Migration):

    initial = True

    dependencies = [
    ]

    operations = [
        migrations.CreateModel(
            name='Category',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('created_at', models.DateTimeField(auto_now_add=True, help_text='The date this record was created.', verbose_name='Create Date')),
                ('updated_at', models.DateTimeField(auto_now=True, help_text='The date this record was updated.', verbose_name='Update Date')),
                ('name', models.CharField(error_messages={'unique': 'A category with that name already exists!'}, help_text='This is the expense category name.', max_length=50, unique=True, validators=[django.core.validators.RegexValidator(code='invalid_category_name', message='Category name must not be blank', regex='[^\\s]+')], verbose_name='Category Name')),
                ('description', models.CharField(help_text='This is the expense category description.', max_length=250, verbose_name='Category Description')),
            ],
            options={
                'db_table': 'category',
                'indexes': [models.Index(fields=['created_at'], name='category_created_74b871_idx'), models.Index(fields=['updated_at'], name='category_updated_830e86_idx')],
            },
        ),
    ]
